File: sdk-dist/kotlin/src/main/kotlin/com/etd/etd.kt
import com.etd.JsonRpcRequest
import com.etd.JsonRpcResponse
import io.ktor.client.*
import io.ktor.client.engine.cio.*
import io.ktor.client.features.json.*
import io.ktor.client.features.json.serializer.*
import io.ktor.client.request.*
import io.ktor.http.*

/**
 * Getd provides several extensions to the standard etd JSON-RPC namespace
 */
class Etd(val url: String) {

    private val client = HttpClient {
        install(JsonFeature) {
            serializer = GsonSerializer()
        }
    }

    /**
     * This method is used for real-time events through subscriptions
     *  See the subscription documentation for more information
     */
    suspend fun subscribe(): Void {

        val response: JsonRpcResponse<Void> = client.post(url) {
            contentType(ContentType.Application.Json)
            body = JsonRpcRequest(method = "eth_subscribe", params = listOf(), jsonrpc = "2.0", id = 1)
        }
        return response.result

    }

    /**
     * This method is used for real-time events through subscriptions
     *  See the subscription documentation for more information
     */
    suspend fun unsubscribe(): Void {

        val response: JsonRpcResponse<Void> = client.post(url) {
            contentType(ContentType.Application.Json)
            body = JsonRpcRequest(method = "eth_unsubscribe", params = listOf(), jsonrpc = "2.0", id = 1)
        }
        return response.result

    }

}

File: sdk-dist/typescript/lib/json_rpc/etd.ts
import axios from "axios";

/**
 * Getd provides several extensions to the standard etd JSON-RPC namespace
 */
export class Etd {
  baseURL: string;
  port?: number;
  url: string;

  constructor(baseURL: string, port?: number) {
    this.baseURL = baseURL;
    this.port = port;
    this.url = port ? `${baseURL}:${port}` : baseURL;
  }
  /**
   * This method is used for real-time events through subscriptions
   *  See the subscription documentation for more information
   */
  async subscribe(): Promise<void> {
    let response = await axios.post(this.url, {
      method: "eth_subscribe",
      params: undefined,
      jsonrpc: "2.0",
      id: 1,
    });

    return response.data.result;
  }

  /**
   * This method is used for real-time events through subscriptions
   *  See the subscription documentation for more information
   */
  async unsubscribe(): Promise<void> {
    let response = await axios.post(this.url, {
      method: "eth_unsubscribe",
      params: undefined,
      jsonrpc: "2.0",
      id: 1,
    });

    return response.data.result;
  }
}
